### PR TITLE
Adds screenless IPCs

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

This adds support for screenless IPCs. Most notably adds the Sprinter IPC chasis (by @Ebin-Halcyon ) to show this off

image

image

Also adds IPC tails

## Why It's Good For The Game

More IPC customization

Also Synths are no longer FBP jank, but who cares? Screenless IPC support!

## Changelog

:cl: Halcyon, retlaw34, thgvr
add: Added the Sprinter IPC Chassis
add: Added the PGF MECHANICS IPC Chassis
add: Adds IPC tails
tweak: Robotic eyes now use the synth eye sprite.
/:cl:

### CODE REVIEW

**Comment by rye-rice** [comment]
drafting because i realized i could add more stuff, please still test merge this

**Comment by thgvr** [comment]
Dw it will be done

**Comment by Sweettoothart** [comment]
Issue: Synth tails are consistent between IPC character slots.

Replication: Choose tail for an ipc, go to another ipc slot, the tail will be the same option as the one previously chosen.

File: code/modules/mob/dead/new_player/sprite_accessories/ipc.dm [hunk 1]
```diff
@@ -232,12 +232,71 @@
...
+	name = "Power Plug"
+	icon_state = "plug"
+	secondary_color = TRUE
+
+/datum/sprite_accessory/ipc_tail/cat
+	name = "Pawsitrons Cat"
+	icon_state = "cat"
+
+/datum/sprite_accessory/ipc_tail/fox
+	name = "Pawsitrons Fox"
```
Comment (thgvr) [inline code/modules/mob/dead/new_player/sprite_accessories/ipc.dm:287]: Don't like these, doesn't feel very ipc to me

Comment (Martinpachu) [reply code/modules/mob/dead/new_player/sprite_accessories/ipc.dm:287]: why not? besides,synths aren't very classic ipc-ey,either

Comment (Hydrosatan) [reply code/modules/mob/dead/new_player/sprite_accessories/ipc.dm:287]: I love them and I've seen a couple IPCs with them. They're very cool

**Comment by rye-rice** [comment]
marge please

File: code/modules/mob/dead/new_player/sprite_accessories/ipc.dm [hunk 2]
```diff
@@ -236,12 +236,71 @@
...
 // Start chassis - the worst thing ever please rework this
 
 /datum/sprite_accessory/ipc_chassis // Used for changing limb icons, doesn't need to hold the actual icon. That's handled in ipc.dm
 	icon = null
 	icon_state = "who cares fuck you" // In order to pull the chassis correctly, we need AN icon_state(see line 36-39). It doesn't have to be useful, because it isn't used.
 	color_src = 0
+	var/use_eyes = FALSE //do we use normal robotic eyes? used when we dont want a screen but still want visible eyes
+	var/has_screen = TRUE //do we have a screen to toggle
+	var/has_overlay = FALSE //does this chasis have a overlay icon?
+	var/is_digi = FALSE //does this chasis use digigrade
```
Comment (MarkSuckerberg) [inline code/modules/mob/dead/new_player/sprite_accessories/ipc.dm]:
```suggestion
	var/is_digi = FALSE //does this chasis use digitigrade
```

File: code/modules/mob/living/carbon/human/species_types/IPC.dm [hunk 3]
```diff
@@ -110,6 +115,11 @@
 	if(!ishuman(owner))
 		return
 	var/mob/living/carbon/human/H = owner
+	var/datum/species/ipc/species_datum = H.dna.species
```
Comment (MarkSuckerberg) [inline code/modules/mob/living/carbon/human/species_types/IPC.dm:120]: Honestly I'd just add the same check to even giving the IPC the action in the first place here:

https://github.com/retlaw34/Shiptest/blob/9c84ad7525ebe5fed6e44f85103a80119b23663c/code/modules/mob/living/carbon/human/species_types/IPC.dm#L70-L72

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/IPC.dm (hunk 3 above)
Comment (rye-rice) [reply code/modules/mob/living/carbon/human/species_types/IPC.dm:120]: should be addressed now, the screen action button is never given now

**Comment by MarkSuckerberg** [comment]
gotta fix whatever this is first
`Limb generated with nonexistant icon. File: icons/mob/species/ipc/bodyparts.dmi | State: pgfipc-d_r_leg`

**Comment by rye-rice** [comment]
Yeahhh, the digi right leg shows up in game, so I have no clue what it could be

**Review by MarkSuckerberg — DISMISSED** [review]
can't merge this until the error's resolved one way or another
the issue is that it's trying to generate the limbs with non-digitigrade icons, so either find a way to fix that or just add plantigrade sprites for the pgf ipc limbs (`pgfipc-d_r_leg` and `pgfipc-d_l_leg`)

**Comment by rye-rice** [comment]
Ohhhhhhhhhh
Oh okay I think I get it now

**Comment by rye-rice** [comment]
> can't merge this until the error's resolved one way or another the issue is that it's trying to generate the limbs with non-digitigrade icons, so either find a way to fix that or just add plantigrade sprites for the pgf ipc limbs (`pgfipc-d_r_leg` and `pgfipc-d_l_leg`)

OOOOH I DID IT, ITS FIXED!!!

I tested in game, digi variant still works

**Pull request merged**